### PR TITLE
EPMRPP-48851: [CLIENT] Javascript CLIENT Add information about agent name, version and other to system attributes

### PR DESCRIPTION
**Pull request by KatsiarynaTatarynovich**
Get information to system attributes from agent:
 * agent name
 * agent version
 * client name
 * client version
 * browser
 * processor type,
 * RAM size
 * OS

The questions
Should this information always be sent or the user decides for himself?
-This info always should be sent

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,6 +27,7 @@
   "dependencies": {
     "axios": "^0.18.1",
     "axios-retry": "^3.1.2",
+    "detect-browser": "^5.0.0",
```
Comment (ElenaRomanchuk) [inline package.json]: According to [description](https://www.npmjs.com/package/detect-browser) this package just return `process.version` for nodeJS-application. I don't think we exactly need it.

File: lib/helpers.js [hunk 2]
```diff
@@ -45,4 +50,30 @@ module.exports = {
...
+        }, {
+            key: 'os',
+            value: `${osType}|${osArchitecture}`,
+            system: true,
+        }, {
+            key: 'RAMSize',
+            value: `${RAMSize}`,
+            system: true,
+        }, {
+            key: 'browser',
```
Comment (ElenaRomanchuk) [inline lib/helpers.js]: I suppose that information about browser should be provided by test runner. Node version could be usefull information, but should be setted to another attribute.

**Review by ElenaRomanchuk — CHANGES_REQUESTED** [review]

File: lib/report-portal-client.js [hunk 3]
```diff
@@ -145,9 +152,15 @@ class RPClient {
             this.map[tempId] = this.getNewItemObj(resolve => resolve(launchDataRQ));
             this.map[tempId].realId = launchDataRQ.id;
         } else {
+            const systemAttr = helpers.getSystemAttribute();
+            const attributes = launchDataRQ.attributes != null
+                && typeof launchDataRQ.attributes[Symbol.iterator] === 'function'
```
Comment (AmsterGet) [inline lib/report-portal-client.js]: Can `attributes` be another iterable object type than array?
May it be just like the next line?
```
const attributes = Array.isArray(launchDataRQ.attributes) ? launchDataRQ.attributes.concat(systemAttr) : systemAttr;
```

Comment (KatsiarynaTatarynovich) [reply lib/report-portal-client.js]: It makes sense but I added this check for additional security.

**Comment by BorisOsipov** [comment]
@KatsiarynaTatarynovich can you copy description of EPMRPP-48851? What is the value of such info in launch attrs? How can I disable this in agent code... how cares about the nodejs version\ram size of code where tests run?

**Comment by KatsiarynaTatarynovich** [comment]
@BorisOsipov added all info in the description. 
@evjlobanova Could you please clarify the need to use this data as system attributes

**Comment by BorisOsipov** [comment]
@KatsiarynaTatarynovich thank you!

File: spec/helper.spec.js [hunk 4]
```diff
@@ -79,8 +74,8 @@ describe('helpers', () => {
                 value: '1',
                 system: true,
             }, {
-                key: 'browser',
-                value: 'browserName|1.0|os',
+                key: 'node',
```
Comment (ElenaRomanchuk) [inline spec/helper.spec.js]: I suggest to rename it to `nodeJS` to avoid confusion with Kubernetes node.

**Review by ElenaRomanchuk — CHANGES_REQUESTED** [review]

**Comment by evjlobanova** [comment]
@BorisOsipov system attributes are invisible on UI. Now we are planning to collect this information. And in future versions we are planning to present this info for users on a widget. 
We have a lot of request regarding this functionality.
We are planning to introduce something common as Allure report have in Env sections
image

**Comment by BorisOsipov** [comment]
@evjlobanova 

>system attributes are invisible on UI.

get it.

> introduce something common as Allure report have in Env sections

nice. thank you for explanation!

**Comment by evjlobanova** [comment]
@BorisOsipov if you want I can share with you  the whole list of features, that we are trying to add to agents.

**Review by ElenaRomanchuk — APPROVED** [review]

**Comment by BorisOsipov** [comment]
@evjlobanova I would like. You can share here, or by email(in github profile), or dm in slack

**Pull request merged**